lib/handlers/connect.js

```diff
@@ -66,9 +67,14 @@ function init (client, packet, done) {
   if (packet.protocolVersion === 3 && clientId.length > client.broker.maxClientsIdLength) {
     returnCode = 2
   }
+  // check if the client keepalive is compatible with broker settings
+  if (client.broker.keepaliveLimit && (!packet.keepalive || packet.keepalive > client.broker.keepaliveLimit)) {
```

Comment:
Shouldn't be packet keepalive `0` allowed?

Reply:
client.keepalive 0 means "infinite keepalive", so if I set a broker.keepaliveLimit, I would like to block the clients with keepalive 0

Reply:
by default, broker.keepaliveLimit is set to 0: this makes no changes in aedes behaviour, because no client.keepalive check is done